uniflow/flow/client.py

```diff
@@ -52,19 +59,76 @@ def __init__(self, config: TransformConfig) -> None:
         """
         self._config = config
         self._server = TransformServer(asdict(self._config))
+        self._encoder = tiktoken.encoding_for_model("gpt-3.5")
 
     def run(self, input_list: List[Mapping[str, Any]]) -> List[Mapping[str, Any]]:
         """
-        Run the flow
+        Process and possibly split input contexts before running them through the server.
+
+        This method checks if auto-splitting of long text is enabled in the configuration. If so, it evaluates each input context.
+        If a context exceeds a certain size, it is split into smaller chunks to meet the token size limitations.
+        Each chunk is then processed separately. If auto-splitting is not enabled, or if the context is within the acceptable size limit,
+        the input is processed as is.
 
         Args:
             input_list (List[Mapping[str, Any]]): List of inputs to the flow
 
         Returns:
             List[Mapping[str, Any]]: List of outputs from the flow
         """
-        # convert config to dict for future remote calls
-        output = self._server.run(input_list)
+
+        processed_input = []
+
+        # Check if auto-splitting of long text is enabled
+        if self._config.auto_split_long_text:
+            # Define the token size limitation
+            token_size_limit = 4096
```

Comment:
nit: shall we move the parameter `token_size_limit` into the class init function?

Reply:
Done